aws/resource_aws_appconfig_deployment.go

```diff
@@ -0,0 +1,216 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/appconfig"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsAppconfigDeployment() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppconfigDeploymentCreate,
+		Read:   resourceAwsAppconfigDeploymentRead,
+		Update: resourceAwsAppconfigDeploymentUpdate,
+		Delete: resourceAwsAppconfigDeploymentDelete,
+
+		Schema: map[string]*schema.Schema{
+			"application_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"environment_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"deployment_strategy_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"configuration_profile_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"configuration_version": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 1024),
+				),
+			},
+			"tags": tagsSchema(),
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"deployment_number": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceAwsAppconfigDeploymentCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).appconfigconn
+	applicationID := aws.String(d.Get("application_id").(string))
+	environmentID := aws.String(d.Get("environment_id").(string))
+	deploymentStrategyID := aws.String(d.Get("deployment_strategy_id").(string))
+	configProfileID := aws.String(d.Get("configuration_profile_id").(string))
+	deploymentDescription := aws.String(d.Get("description").(string))
+	configVersion := aws.String(d.Get("configuration_version").(string))
+
+	input := &appconfig.StartDeploymentInput{
+		ApplicationId:          applicationID,
+		EnvironmentId:          environmentID,
+		DeploymentStrategyId:   deploymentStrategyID,
+		ConfigurationProfileId: configProfileID,
+		ConfigurationVersion:   configVersion,
+		Description:            deploymentDescription,
+		Tags:                   keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().AppconfigTags(),
+	}
+
+	var deploy *appconfig.StartDeploymentOutput
+	var err error
+	err = resource.Retry(2*time.Minute, func() *resource.RetryError {
+		deploy, err = conn.StartDeployment(input)
+		if err != nil {
+			if isAWSErr(err, appconfig.ErrCodeConflictException, "") {
+				log.Printf("[DEBUG] Retrying AppConfig Deployment creation: %s", err)
```

Comment:
remove debug print statements?

Reply:
There's other debug logs when retrying (like [this](https://github.com/figma/terraform-provider-aws/blob/f7edb12cc2d419206ed7a51a827c32936331fbc3/aws/resource_aws_backup_selection.go#L108)). It might be a pattern to keep them in.